src/app/[locale]/admin/revenue/components/RevenueFilter.tsx

```
@@ -33,40 +48,155 @@ interface RevenueFilterProps {
+    const handleWeeklyQuickFilter = (preset: 'thisWeek' | 'lastWeek' | 'last4Weeks') => {
+        const today = startOfToday();
+        let weekStart: Date;
+        let weekEnd: Date;
+
+        switch (preset) {
+            case 'thisWeek':
+                weekStart = startOfWeek(today, { weekStartsOn: 1 });
+                weekEnd = endOfWeek(today, { weekStartsOn: 1 });
+                break;
+            case 'lastWeek':
+                const lastWeek = subWeeks(today, 1);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)